controllers/cdapmaster_controller.go

```diff
@@ -133,9 +133,20 @@ func ApplyDefaults(resource interface{}) {
 		spec.Config[confLocalDataDirKey] = confLocalDataDirVal
 	}
 
-	// Set twill.security.secret.disk.name to be consistent with securitySecret if not overwritten.
-	if _, ok := spec.Config[confTwillSecuritySecretDiskName]; !ok {
-		spec.Config[confTwillSecuritySecretDiskName] = spec.SecuritySecret
+	// Set security secret disk names to be consistent with securitySecret if not overwritten.
+	if _, ok := spec.Config[confTwillSecurityMasterSecretDiskName]; !ok {
```

Comment:
nit: just to avoid any confusion, wrap this within

if len(spec.SecuritySecret) > 0 {
  // override 
}

Reply:
We only want to override if the cdap-cr doesn't already have that config set right?

Reply:
Yes. 

But if security secret is empty in cdap-cr, we don't need to set these configures to empty again, right?

Reply:
Fixed, thanks!